lib/src/generated_serialization_rules.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/**
+ * This is a template file that will be copied to the projects that use
+ * transformer-based serialization and will be filled with the generated
+ * serialization rules.
+ */
+library generated_serialization_rules;
+
+// ====== Imports generated by the transformer will be appended below =======
+// ======== Do not remove the {{GENERATED_IMPORTS_MARKER}} comment ==========
+
+import 'package:serialization/serialization.dart' as s show Serialization;
+import 'package:serialization/serialization.dart' hide Serialization;
+export 'package:serialization/serialization.dart';
+//{{GENERATED_IMPORTS_MARKER}}
```

Comment:
If we don't expect the user to modify this file (which is good) then I don't think we need to use special markers and replace them, we can just generate this string directly using interpolation or a StringBuffer, which would be simpler. It makes it a bit harder to reason about this file initially, but if we have something that can generate it empty and have it there for the user to see that should be much easier.